adapters/mqtt-vertx/src/main/java/org/eclipse/hono/adapter/mqtt/impl/MessageMapping.java

```diff
@@ -0,0 +1,111 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.eclipse.hono.adapter.mqtt.impl;
+
+import org.eclipse.hono.adapter.mqtt.MqttContext;
+import org.eclipse.hono.adapter.mqtt.MqttProtocolAdapterProperties;
+import org.eclipse.hono.config.MapperEndpoint;
+import org.eclipse.hono.util.MessageHelper;
+import org.eclipse.hono.util.RegistrationConstants;
+import org.eclipse.hono.util.ResourceIdentifier;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import io.vertx.core.Future;
+import io.vertx.core.MultiMap;
+import io.vertx.core.Promise;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.impl.headers.VertxHttpHeaders;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.client.HttpResponse;
+import io.vertx.ext.web.client.WebClient;
+import io.vertx.mqtt.messages.MqttPublishMessage;
+import io.vertx.mqtt.messages.impl.MqttPublishMessageImpl;
+
+/**
+ * A Vert.x based message mapping component.
+ */
+@Component
+public class MessageMapping {
+
+    private WebClient webClient;
+
+    private WebClient getWebClient() {
+        if (webClient == null) {
+            webClient = WebClient.create(null);
+        }
+        return webClient;
+    }
+
+    /**
+     * Sets the vertx webClient to use when mapping payload.
+     *
+     * @param webClient the webClient to set
+     */
+    @Autowired
+    public void setWebClient(final WebClient webClient) {
+        this.webClient = webClient;
+    }
+
+    Future<MappedMessage> mapMessage(final MqttContext ctx, final ResourceIdentifier targetAddress,
```

Comment:
public ?
could also use a proper JavaDoc comment

Reply:
Fixed that